include/deal.II/multigrid/mg_coarse.h

```diff
@@ -329,6 +339,22 @@ MGCoarseGridApplySmoother<VectorType>::initialize (const MGSmootherBase<VectorTy
     (&coarse_smooth_,typeid(*this).name());
 }
 
+
+template<class VectorType>
+void
+MGCoarseGridApplySmoother<VectorType>::clear()
+{
+  coarse_smooth = nullptr;
+}
+
+
+template<class VectorType>
+MGCoarseGridApplySmoother<VectorType>::~MGCoarseGridApplySmoother()
+{
+  clear();
```

Comment:
for convenience, i would say. Note that there is the same destructor for `MGCoarseGridLACIteration`, so i just made it the same.  One could of course skip it and just call `clear()` in the destructor of the class which encapsulates `MGCoarseGridApplySmoother` .